spec/features/user_submits_new_location_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+feature 'user submits a new location', %Q{
+  As a signed in user
+  I want to submit a new location
+  To contribute to the locations database
+} do
+
+  scenario 'signed in user submits a new location successfully' do
+    user = User.create(
+      first_name: "John",
+      last_name: "Smith",
+      email: "jsmith@email.com",
+      password: "password")
+
+    visit new_user_session_path
+
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+
+    click_button 'Log in'
+
+    click_link 'Locations'
+
+    visit new_location_path
+
+    fill_in 'Location Name', with: 'Chipotle'
+    fill_in 'Address', with: 'Summer St'
+    fill_in 'City', with: 'Boston'
+    fill_in 'State', with: 'MA'
+    fill_in 'Zip Code', with: '02111'
+
+    click_button 'Create Location'
+
+    expect(page).to have_content('New Location Added')
+  end
+
+  scenario 'signed in user submits an invalid location' do
+    user = User.create(
+      first_name: "John",
+      last_name: "Smith",
+      email: "jsmith@email.com",
+      password: "password")
+
+    visit new_user_session_path
+
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
```

Comment:
Check out the Capybara method sign_in_as to avoid having to do all this setup in your test :)